### PR TITLE
[AEG-10] Connect backend APIs with add/edit health record screen

### PR DESCRIPTION
**Pull request by pointmekin**
# Notes

Connected APIs
- Fetching table data for the specified health record. (Checked for elderly users)
- Add health record entry to the specified hrName of the user (Checked for elderly users)
- Edit cover image of the user's specified health record (Checked for elderly users)
- Delete health record (Not checked yet, awaiting create health record functionality before testing)

Front-end
- Connected the Health Record selection to the Add/Edit health record screen.
- Created relevant confirmation, error, and success alerts

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -88,16 +89,18 @@ const App = () => {
...
-                {...{
-                  borderRadius: 4,
-                  labels: tourGuideLabels,
-                  androidStatusBarVisible: true
-                }}>
-                <NativeBaseProvider theme={theme}>
-                  <MainNavigation />
-                </NativeBaseProvider>
-              </TourGuideProvider>
+              <HaelthRecordContextProvider>
```
Comment (chuwong35122) [inline App.tsx]: typo

File: src/contexts/HealthRecordContext.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    try {
+      const tempHrName = currentHrName;
+      const tempElderlyUid = currentElderlyUid;
+      const { data } = await client.get(
+        `healthRecord/table/${
+          user.isElderly ? '' : `${tempElderlyUid}/`
+        }${tempHrName}`
+      );
+      const _data = { ...data, data: data.data.reverse() };
+      console.log(_data);
```
Comment (chuwong35122) [inline src/contexts/HealthRecordContext.tsx]: remove

File: src/screens/HealthRecordingsScreen.tsx [hunk 3]
```diff
@@ -83,7 +92,16 @@ const HealthRecordingsScreen = () => {
     });
   }, [eventEmitter]);
 
-  // -----------------------
+  useFocusEffect(
```
Comment (chuwong35122) [inline src/screens/HealthRecordingsScreen.tsx:95]: can you comment why you have this function?

File: src/screens/healthRecord/AddHealthEntryScreen.tsx [hunk 4]
```diff
@@ -1,58 +1,111 @@
...
   const {
     control,
     formState: { errors },
-    handleSubmit
+    handleSubmit,
+    watch,
+    reset
   } = useForm({ mode: 'onTouched' });
 
+  // States
```
Comment (chuwong35122) [inline src/screens/healthRecord/AddHealthEntryScreen.tsx]: remove?

File: src/screens/healthRecord/AddHealthEntryScreen.tsx [hunk 5]
```diff
@@ -1,58 +1,111 @@
...
+        }`,
+        payload
+      );
+      if (data) {
+        reset();
+        getHealthRecordTable();
+        setShowSuccessAdd(true);
+      }
+    } catch (error) {
+      // @ts-ignore
```
Comment (chuwong35122) [inline src/screens/healthRecord/AddHealthEntryScreen.tsx]: import  { AxiosError } from 'axios';

try using const err = error as AxiosError;

File: src/screens/healthRecord/EditHealthEntryScreen.tsx [hunk 6]
```diff
@@ -21,65 +25,171 @@ const EditHealthEntryScreen = ({
...
-  const {
-    control,
-    formState: { errors },
-    handleSubmit
-  } = useForm({ mode: 'onTouched' });
+  const { handleSubmit } = useForm({ mode: 'onTouched' });
   const { ScreenWidth } = useDimensions();
   const { takePicture, selectPictureFromDevice } = useImageSelection();
 
+  // States
```
Comment (chuwong35122) [inline src/screens/healthRecord/EditHealthEntryScreen.tsx]: remove?

**Review by chuwong35122 — CHANGES_REQUESTED** [review]

File: src/screens/HealthRecordingsScreen.tsx (hunk 3 above)
Comment (pointmekin) [reply src/screens/HealthRecordingsScreen.tsx:95]: will update Readme

**Review by chuwong35122 — APPROVED** [review]

**Pull request merged**